xclim/sdba/processing.py

```diff
@@ -527,22 +778,29 @@ def stack_variables(ds, rechunk=True, dim="variables"):
     Returns
     -------
     xr.DataArray
+      The transformed variable. Attributes are conserved, even if some might be incorrect.
+      Except units, which are replace with "". Old units are stored in `sdba_transformation_units`.
+      A `sdba_transform` attribute is added, set to the transformation method.
+      `sdba_transform_lower` and `sdba_transform_upper` are also set if the requested bounds are different from the defaults.
+
       Array with variables stacked along `dim` dimension. Units are set to "".
+      Variables are sorted alphabetically.
     """
     # Store original arrays' attributes
     attrs = {}
-    nvar = len(ds.data_vars)
-    for i, var in enumerate(ds.data_vars.values()):
+    datavars = sorted(ds.data_vars.items(), key=lambda e: e[0])
```

Comment:
Why the sorted ?

Reply:
I'll add a note in the doc.
It's to make sure that if you call `stack_variables` on two different datasets with the same variables, the order of the new `variables` dimension is the same on both new `DataArrays`.

I think this also needs a check in the adjustment object.